nixos/modules/services/networking/StevenBlack-hosts.nix

```diff
@@ -0,0 +1,44 @@
+{config, lib, pkgs, ...}:
+with lib;
+{
+  options.services.stevenBlackHost = {
+    enable = mkEnableOption "filtering hosts via StevenBlack's hosts file";
+    flags = mkOption {
+      description = ''
+        Flags to passs while building the hosts file.
+        Note that compress and skipstatichosts are passed by default.
+        See also <link xlink:href="https://github.com/StevenBlack/hosts#command-line-options>here</link> for details.
+        Note that options thet modify directly the hosts file or the global DNS cache will error out as the hosts file is first built in the standard nix sandbox.
+      '';
+      default = {};
+      #TODO: cli setting type
+      type = with types; attrsOf (
+        oneOf [bool str (listOf bool) (listOf str)]
+      );
+      
+    };
+    whitelist = mkOption {
+      description = "List of hosts to allow";
+      default = "";
+      type = types.lines;
+
+    };
+  };
+  
+  config =
+    let
+      cfg = config.services.stevenBlackHost;
+    in mkIf cfg.enable {
+      
+      services.stevenBlackHost.flags={
+        compress = mkDefault true;
+        skipstatichosts = mkDefault true;
+      };
+      
+      networking.extraHosts = 
+        builtins.readFile (pkgs.StevenBlack-hosts.override{
```

Comment:
This is a bit ugly here, I think this is IFD (import-from-derivation), where a derivation needs to be built during evaluation time. I just created https://github.com/NixOS/nixpkgs/pull/81945 to get around this, after which you will be able to do
```nix
{
  networking.hostFiles = [
    (pkgs.StevenBlack-hosts.override { /* ... */ })
  ];
}
```

Reply:
Just merged above PR, so this can be done now